api/app/Rules/SkillLevelRequiredIfEssential.php

```diff
@@ -54,8 +54,8 @@ public function validate(string $attribute, mixed $incomingLevel, Closure $fail)
         if (! is_null($templateId)) {
             $template = JobPosterTemplate::findOrFail($templateId);
             $model = $template->skills()->firstWhere('skill_id', $skillId);
-            $currentType = $model->pivot->type;
-            $currentLevel = $model->pivot->required_skill_level;
+            $currentType = $model->getAttributeValue('type');
+            $currentLevel = $model->getAttributeValue('required_skill_level');
```

Comment:
This is to avoid it since `pivot` is a magic getter. I can add it in the PHPDoc for `PoolSkill` but that also seemed weird :thinking: